src/key-casing.ts

```diff
@@ -38,7 +38,26 @@ function deepTransformKeys<T>(obj: T, transform: (s: string) => string): T {
 }
 
 /**
- * Transforms the keys of an Record to camelCase.
+ * This function is used to shallowly transform the keys of an object.
+ * It will only be transformed at runtime, so it's not type safe.
+ * @param obj the object to transform.
+ * @param transform the function to transform the keys from string to string.
+ * @returns the transformed object.
+ * @example transformKeys({ 'foo-bar': { 'fizz-buzz': true } }, toCamelCase)
+ * // { fooBar: { 'fizz-buzz': true } }
+ */
+function transformKeys<T>(obj: T, transform: (s: string) => string): T {
```

Comment:
I would like to see more-specific typing here. I think the transform function could be made generic to accept a "transform" type that would yield more specific output types. Let me think on this